val.py

```diff
@@ -302,108 +322,113 @@ def run(
             if single_cls:
                 pred[:, 5] = 0
             predn = pred.clone()
+            predntwee = pred.clone()  # TODO can we use predn, or are these values changed?
```

Comment:
I checked and `scale_boxes` modifies `predn`, so do not use it. 

But can you rename it to something more meaningful?